dbt_semantic_interfaces/transformations/measure_to_metric_transformation_pieces/measure_features_to_metric_name.py

```diff
@@ -94,13 +97,69 @@ def _metrics_equivalent(search_metric: PydanticMetric, manifest_metric: Pydantic
                 return existing_metric
         return None
 
+    @staticmethod
+    def update_required_measure_features_in_simple_model(
+        *,
+        measure: PydanticMeasure,
+        semantic_model_name: str,
+        metric: PydanticMetric,
+        # Measure input fields
+        fill_nulls_with: Optional[int],
+        join_to_timespine: Optional[bool],
+        measure_input_filters: Optional[PydanticWhereFilterIntersection],
+    ) -> None:
+        """Set the measure features on the metric, as appropriate.
+
+        Use this when merging an existing measure's
+        arguments into a metrics.
+
+        This will update the metric in place rather than returning a new one.
+        """
+        assert metric.type == MetricType.SIMPLE, "Attempted to set measure features on a non-simple metric"
+        if metric.type_params.metric_aggregation_params is not None:
+            # these values have already been set.
+            return
+
+        # We only set these if they are passed in explicitly so we can avoid overriding defaults.
+        if fill_nulls_with is not None:
+            metric.type_params.fill_nulls_with = fill_nulls_with
+        if join_to_timespine:
+            metric.type_params.join_to_timespine = join_to_timespine
+
+        metric.type_params.metric_aggregation_params = PydanticMetric.build_metric_aggregation_params(
+            measure=measure,
+            semantic_model_name=semantic_model_name,
+        )
+        # Measures without an expr fall back to using the measure name as the column name,
+        # so we need to enable mimicking that behavior here.
+        if metric.type_params.expr is None:
+            metric.type_params.expr = measure.expr or measure.name
+
+        filters = metric.filter.where_filters if metric.filter else []
```

Comment:
I think if you reverse the order of how these filters are added to the list (i.e. measure filters first), it'll match the existing behavior better.